pkg/controller/container-runtime-config/container_runtime_config_controller.go

```diff
@@ -873,6 +886,90 @@ func registriesConfigIgnition(templateDir string, controllerConfig *mcfgv1.Contr
 	return &registriesIgn, nil
 }
 
+func (ctrl *Controller) createAddInheritableCapabilitiesMC() error {
+	var configMapName = "crio-add-inheritable-capabilities"
+
+	// Check if the crio-add-inheritable-capabilities config map exists in the openshift-machine-config-operator namespace
+	addInheritableCapabilitiesCM, err := ctrl.kubeClient.CoreV1().ConfigMaps(ctrl.namespace).Get(context.TODO(), configMapName, metav1.GetOptions{})
+	capabilitiesCMIsNotFound := errors.IsNotFound(err)
+	if err != nil && !capabilitiesCMIsNotFound {
+		return fmt.Errorf("error checking for %s config map: %v", configMapName, err)
+	}
+	// If the crio-add-inheritable-capabilities config map exists, that means the crio-add-inheritable-capabilities MC was already created
+	// so we should not create this MC again and return
+	if addInheritableCapabilitiesCM != nil && !capabilitiesCMIsNotFound {
+		return nil
+	}
+
+	sel, err := metav1.LabelSelectorAsSelector(metav1.AddLabelToSelector(&metav1.LabelSelector{}, builtInLabelKey, ""))
+	if err != nil {
+		return err
+	}
+	// Find all the MachineConfigPools
+	mcpPoolsAll, err := ctrl.mcpLister.List(sel)
+	if err != nil {
+		return err
+	}
+
+	// Create the crio-add-inheritable-capabilities MC for all the available pools
+	for _, pool := range mcpPoolsAll {
+		managedKey := getManagedKeyCapabilities(pool)
+		mc, err := ctrl.client.MachineconfigurationV1().MachineConfigs().Get(context.TODO(), managedKey, metav1.GetOptions{})
+		isNotFound := errors.IsNotFound(err)
+		if err != nil && !isNotFound {
+			return fmt.Errorf("error checking for %s machine config: %v", managedKey, err)
+		}
+		// continue to the next MC if this already exists
+		if mc != nil && !isNotFound {
+			continue
+		}
+
+		tempIgnCfg := ctrlcommon.NewIgnConfig()
+		mc, err = ctrlcommon.MachineConfigFromIgnConfig(pool.Name, managedKey, tempIgnCfg)
+		if err != nil {
+			return fmt.Errorf("could not create crio-add-inheritable-capabilities MachineConfig from new Ignition config: %v", err)
+		}
+		rawCapsIgnition, err := json.Marshal(createNewIgnition(createDefaultAddInheritableCapabilitiesFile()))
+		if err != nil {
+			return fmt.Errorf("error marshalling crio-add-inheritable-capabilities config ignition: %v", err)
+		}
+		mc.Spec.Config.Raw = rawCapsIgnition
+		// Create the crio-add-inheritable-capabilities MC
+		if err := retry.RetryOnConflict(updateBackoff, func() error {
+			_, err = ctrl.client.MachineconfigurationV1().MachineConfigs().Create(context.TODO(), mc, metav1.CreateOptions{})
```

Comment:
gonna be honest, I don't really know, @QiWang19 do you know :)

Reply:
To clarify, if this does have a generated-by-controller-version, it would have an issue in future upgrades. It must be created as a "user MC" without controller control after generation